PipeMethodCalls/Extensions/NamedPipeServerStreamExtensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.IO.Pipes;
+
+namespace PipeMethodCalls
+{
+    /// <summary>
+    /// Extensions to make upgrading and working with NamedPipeServerStream easier.
+    /// </summary>
+    public static class NamedPipeServerStreamExtensions
+    {
+        /// <summary>
+        /// Method of quickly upgrading an existing <see cref="NamedPipeServerStream"/> into a <see cref="PipeServerWithCallback{TRequesting,THandling}"/>.
+        /// </summary>
+        /// <param name="existingPipe">Instance of a named pipe to upgrade.</param>
+        /// <param name="handlerFactoryFunc">A factory function to provide the handler implementation.</param>
+        /// <typeparam name="TRequesting">The callback channel interface that the client will be handling.</typeparam>
+        /// <typeparam name="THandling">The interface for requests that this server will be handling.</typeparam>
+        /// <returns></returns>
+        public static PipeServerWithCallback<TRequesting, THandling> UpgradeWithCallback<TRequesting, THandling>(
```

Comment:
I don't think we need the extension methods here. The constructors are plenty usable and discoverable.

Reply:
Understandable, I can remove these.